enclave/core/sgx/enter.S

```diff
@@ -39,20 +39,50 @@
 oe_enter:
 .cfi_startproc
 
+.get_td:
+
+    // Upon first entry to the enclave, td->base.self in the td_t structure
+    // is not yet initialed. However, the loader in host/sgx/create.c places
+    // the td_t structure as a specific offset from TCS, which we can obtain
+    // via td_from_tcs
+
+    // Save register state to stack
+    push %rax
+    push %rbx
+    push %rcx
+    push %rdi
+    push %rsi
+    push %rdx
+
+    // Get the location of the td_t structure for this thread. This value is
+    // expected to be present in %r11 for the remainder of oe_enter.
+    mov %rbx, %rdi
+    call td_from_tcs
```

Comment:
td_from_tcs just adds a known offset to tcs to obtain td_t. Can we just do the addition here instead of calling a function?